### PR TITLE
Example basic acceptance tests work with chef-acceptance

### PR DESCRIPTION
**Pull request by patrick-wright**
PR for peer review.  The PR represents some experimental work for running Chef acceptance tests.

### CODE REVIEW

File: acceptance/Gemfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+source 'https://rubygems.org'
+
+gem 'chef-acceptance', github: 'chef/chef-acceptance'
+gem 'berkshelf'
```
Comment (sersut) [inline acceptance/Gemfile]: Do we need to add all of these stuff into the Gemfile? I think we are launching `kitchen` via CLI so these doesn't necessarily need to be gem dependencies. We can assume that the runner of `chef-acceptance` has installed Chef DK.

**Comment by sersut** [comment]
This is all good @patrick-wright. But I do not think we want to move the kitchen-tests into acceptance. I think we want to copy them. We can leave it to the `chef` project to decide if those tests should continue to run in travis or not. Let's restore the `kitchen-tests` directory as it is and copy the files we need into chef-acceptance to provide an example to write a test with cookbooks.

File: acceptance/Gemfile (hunk 1 above)
Comment (patrick-wright) [inline acceptance/Gemfile]: My assumption is that chefdk will be installed.  Which I think is a good thing to enforce for acceptance.  We can always add them here if we run into issues.  I'll add that to the README.

**Comment by patrick-wright** [comment]
@sersut I restored `kitchen-tests`.

File: acceptance/Gemfile (hunk 1 above)
Comment (sersut) [inline acceptance/Gemfile]: :+1:

**Comment by sersut** [comment]
:+1: for me. 

@chef/client-core This is the very first example of the acceptance tests that we are going to hook into chef-acceptance. 

Do you see any side effects of introducing this directory?

**Comment by lamont-granquist** [comment]
This seems to duplicate the purpose of kitchen-tests so we should merge them into one directory and run them both from travis and delivery.  Having two of these just means that one or both of them will be kinda shitty depending on which one gets attention.

File: acceptance/.kitchen.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+---
+driver:
+  name: vagrant
+  customize:
+    cpus: 4
+    memory: 2048
+
+provisioner:
+  name: chef_zero
```
Comment (lamont-granquist) [inline acceptance/.kitchen.yml]: Don't you need to use chef_github here and kitchen-appbundle-updater gem in order to run the PR against the new chef code?  If we're assuming that the build has to go through all the way and generate an omnibus artifact, then I'd still expect to see configuration here that would pull that build out of artifactory (hooks in test-kitchen that I do not think exist yet).

File: acceptance/Gemfile (hunk 1 above)
Comment (lamont-granquist) [inline acceptance/Gemfile]: Per other comment would like to see all this duplicated code shared with the travis tests, so we should not make any assumptions about ChefDK being installed.

**Comment by sersut** [comment]
@lamont-granquist our aim here is to write an example test that will be executed under the context of `chef-acceptance` framework. In the future this framework will be hooked into our CI so that we can run the acceptance tests we are running locally in the CI. 

These tests will not be testing against a PR commit. They will be testing against full omnibus artifacts. Hence we are not targeting the github driver. They will be able to pull the version to be tested from any channel (:stable, :current or :unstable when it is ready). 

I think leaving `kitchen-tests` is the right way to go until we start executing this suite in CI. 

Ping me and we can have a chat if there are still concerns :smile:

**Comment by lamont-granquist** [comment]
There still seems to be considerable overlap in the cookbooks then.  Can we just have two different kitchen files like the .kitchen.yml and kitchen.cloud.yml pattern in the community cookbooks?

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by sersut** [comment]
@lamont-granquist overlap is OK for now. I think the contents of this directory will change in the future. As we evolve chef-acceptance in the future, `.kitchen.yml` may even go away and everything might be driven by `.acceptance.yml`. We will be developing chef-acceptance iteratively and the tests we have copied over there are just examples for now.

**Comment by thommay** [comment]
OK, what I'd like to see here is an approach that generates a set of cookbooks that we can use for acceptance tests across the board - both for travis to run against PRs and for delivery/manhattan to run at the test phase of omnibus pipelines.

**Comment by patrick-wright** [comment]
@lamont-granquist @sersut I've removed the copied kitchen-tests from acceptance and left in the basic specs.  Our primary goal is to validate the acceptance framework which we can do without the additional tests.  @thommay We're less concerned with the actual tests and more concerned with validating the acceptance structure.  The idea is to create a few examples of how to setup the acceptance tests.  Hope that clears it up a bit.

File: acceptance/Gemfile (hunk 1 above)
Comment (danielsdeleo) [inline acceptance/Gemfile]: Just curious, could this be done with policyfiles instead? :dog: :cake:

**Comment by sersut** [comment]
I think that's OK for now @patrick-wright. @lamont-granquist / @thommay let us know if this looks good. 

To reiterate what we are trying to do is to prove the infrastructure to be able to write a set of acceptance tests that will work with our internal CI and from developer laptops. Our intention is not to determine and write those tests.

**Comment by sersut** [comment]
Had a conversation with @patrick-wright and @lamont-granquist and we will not merge this right now. We will revisit once the rest of the chef-acceptance is in place.

**Pull request closed without merging**